cnd/src/swap_protocols/rfc003/mod.rs

```diff
@@ -42,3 +45,15 @@ pub enum Error {
     #[error("internal error: {0}")]
     Internal(String),
 }
+
+pub fn alpha_expiry_has_passed<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>(
+    request: &Request<AL, BL, AA, BA>,
+) -> bool {
+    request.alpha_expiry < Timestamp::now()
```

Comment:
:mega: This compares the expiry time to the current system time and not to the actual blockchain time of this ledger (which is used to determine whether or not the generated TX can go in). Not sure if we want to implement this in this fashion.

Reply:
Ok, @bonomat has schooled me on blockchain times and I agree that we need to hit the chain to get the time - I am also of the opinion, along with @D4nte, that this is an optimisation so we can do it later _after_ this PR actually works.  ATM this doesn't work because it breaks current e2e tests and has no added e2e test.  Also, I'm not convinced that we should even be doing this issue.  I'm putting this PR on ice until it can be discussed in person in the office (probably after Christmas)

Reply:
Ok, I will close it for now to remove clutter from the PR page :)